internal/controlplane/handlers_repositories.go

```diff
@@ -99,6 +98,62 @@ func (s *Server) RegisterRepository(
 	}, nil
 }
 
+func (s *Server) repoCreateInfoFromUpstreamRepositoryRef(
+	ctx context.Context,
+	projectID uuid.UUID,
+	providerName string,
+	rep *pb.UpstreamRepositoryRef,
+) (*properties.Properties, *db.Provider, error) {
+	// If the repo owner is missing, GitHub will assume a default value based
+	// on the user's credentials. An explicit check for owner is left out to
+	// avoid breaking backwards compatibility.
+	if rep.GetName() == "" {
+		return nil, nil, util.UserVisibleError(codes.InvalidArgument, "missing repository name")
+	}
+
+	fetchByProps, err := properties.NewProperties(map[string]any{
+		properties.PropertyUpstreamID: fmt.Sprintf("%d", rep.GetRepoId()),
+		properties.PropertyName:       fmt.Sprintf("%s/%s", rep.GetOwner(), rep.GetName()),
```

Comment:
we'll only ever use this for github and that's why we can hardcode the github style name right?

Reply:
uhm... I could probably use the `GetEntityName` function here. But yeah, this is applicable to anything that's compatible with our old repositories protobufs.

Reply:
Switched to using the internal function instead.